### DISCUSSION TITLE
Addition of complex numerical types

### DISCUSSION THREAD

**Discussion by codeflight1** [nasa/fprime#1351] [Ideas]
I was thinking of adding some complex numerical types (e.g. Matrix, Vector, Quaternion) to F`, but was unsure if these fall within the scope of the project. If they do, I would be happy to add them!

**Comment by timcanham** [comment]
It's an interesting idea, but we try not to get too domain specific since different practitioners have different views on how these are represented. It might be a good candidate for a repo of your own you could share with others in the community. I'll leave it open for @LeStarch to comment differently if he would like.

**Reply by LeStarch** [reply]
I agree here.  I think the best place for something like this to start is a separate repo that can be included by project. Then, when traction is gained across multiple project, we can consider it for addition into the core repo.

**Reply by codeflight1** [reply]
That makes sense, thanks for your responses!
